src/shogun/base/SGObject.cpp

```diff
@@ -471,6 +478,68 @@ void CSGObject::save_serializable_post() throw (ShogunException)
 	m_save_post_called = true;
 }
 
+void CSGObject::saveBinary(char* filename) const
+{
+	std::ofstream os(filename);
+	cereal::BinaryOutputArchive archive(os);
+	archive(*this);
+}
+
+void CSGObject::saveJSON(char* filename) const
```

Comment:
@karlnapf that was my first idea as well, but then think about all those interpreted languages, like python :)
```from modshogun import Class, EnumForSerialization
a = Class()
a.save("to_file.json", EnumForSerialization.JSON)
```
seems a bit more shitty than
```from modshogun import Class
a = Class()
a.save_json("to_file.json")
```
or?